stablehlo/dialect/StablehloOps.td

```diff
@@ -1381,7 +1382,8 @@ def StableHLO_AllGatherOp : StableHLO_Op<"all_gather",
 
 def StableHLO_AllReduceOp : StableHLO_Op<"all_reduce",
     [HLO_SpeculatableIfStaticDimInOutputIsStaticInInput,
-     InferTensorType /*all_reduce_c6, all_reduce_c7*/]> {
+     InferTensorType /*all_reduce_c6, all_reduce_c7*/,
```

Comment:
`HLO_PairwiseSameOperandAndResultType`?

Reply:
Looks like we've done this in `infer` to set the inferred element type. That works too

Reply:
Yes, `infer` is taking care of this.